### PR TITLE
Fix ref links in titles

### PR DESCRIPTION
**Pull request by gribnoysup**
Fixes https://github.com/developit/snarkdown/issues/16

Tested both discussed solutions and "global" solution is another one byte smaller, so I went with it 😄

### CODE REVIEW

**Comment by developit** [comment]
Your commitment to size and simplicity is inspiring 😊

File: src/index.js [hunk 1]
```diff
@@ -20,13 +20,14 @@ function encodeAttr(str) {
 	return (str+'').replace(/"/g, '&quot;').replace(/</g, '&lt;').replace(/>/g, '&gt;');
 }
 
+let links = {};
```
Comment (developit) [inline src/index.js]: I think one issue with the global version is that it won't get cleaned after each outer `parse()` call.  So links declared in one document would be available as references in another.

Are you able to push the alternate contextual version?

Comment (gribnoysup) [inline src/index.js]: Yes, of course!

**Comment by developit** [comment]
Not too bad, just a 6 byte increase. Looks good!

**Pull request merged**